react-components/src/query/core-dm/usePointCloudVolumeMappingForAssetInstances.ts

```diff
@@ -80,3 +87,56 @@ export const usePointCloudVolumeMappingForAssetInstances = (
       !isLoading
   });
 };
+
+export const usePointCloudVolumeMappingForIntersection = (
+  intersection: AnyIntersection | undefined
+): UseQueryResult<PointCloudVolumeAssetWithViews[]> => {
+  const fdmSdk = useFdmSdk();
+  const isPointCloudIntersection =
+    intersection?.type === 'pointcloud' &&
+    intersection.volumeMetadata !== undefined &&
+    'assetRef' in intersection.volumeMetadata &&
+    'volumeInstanceRef' in intersection.volumeMetadata;
+  const assetInstanceRefs = useMemo<DMInstanceRef[]>(() => {
+    if (isPointCloudIntersection) {
+      const assetRef = intersection.volumeMetadata?.assetRef;
+      if (assetRef !== undefined && 'externalId' in assetRef) {
+        return [assetRef as DMInstanceRef];
+      }
+    }
+    return [];
+  }, [isPointCloudIntersection, intersection]);
+
+  const { data: volumeMappings, isLoading } =
+    usePointCloudVolumeMappingForAssetInstances(assetInstanceRefs);
+
+  return useQuery({
+    queryKey: [
+      queryKeys.pointCloudDMVolumeAssetMappingsWithViews(
+        assetInstanceRefs
+          .map((assetInstance) => `${assetInstance.externalId}/${assetInstance.space}`)
+          .sort()
+      )
+    ],
+    queryFn: async () => {
+      if (volumeMappings === undefined || volumeMappings.length === 0) {
+        return EMPTY_ARRAY;
+      }
+      const result: PointCloudVolumeAssetWithViews[] = await Promise.all(
+        volumeMappings.map(async (volumeMapping) => {
+          const assetInstance = {
+            externalId: volumeMapping.asset.externalId,
+            space: volumeMapping.asset.space
+          };
```

Comment:
Can we just use `volumeMapping.asset` instead of creating this new object?

Reply:
`volumeMapping.asset` contains all asset properties not just `externalId` and `space`